datadog-agent.yaml

```diff
@@ -218,6 +218,22 @@ subpackages:
               output: fakeintake
               subpackage: "true"
 
+  # The datadog-agent image uses a deprecated 2 year old version of s6-overlay
+  # we don't want to maintain a package for. So instead, include
+  # the raw release of that version of s6-overlay as a
+  # datadog-agent subpackage, only to be used by image builders.
+  - name: datadog-agent-s6-overlay
+    description: "Deprecated s6-overlay for datadog-agent"
+    options:
+      no-provides: true
+    pipeline:
+      - runs: |
+          S6_VERSION=v2.2.0.3
+          JUST_CONTAINERS_DOWNLOAD_LOCATION=https://github.com/just-containers/s6-overlay/releases/download
+          S6ARCH=$([ "$TARGETARCH" = "amd64" ] && echo "amd64" || echo "aarch64") && curl -L ${JUST_CONTAINERS_DOWNLOAD_LOCATION}/${S6_VERSION}/s6-overlay-${S6ARCH}.tar.gz -o s6.tgz
```

Comment:
Uh, this is kind of yucky...  At the very least, I think we need to verify a checksum of this download against a known good checksum included here in this packaging...

Reply:
good call, added a checksum validation step 🙏